src/ImageSharp.Web.Providers.AWS/AmazonS3ClientFactory.cs

```diff
@@ -33,7 +32,7 @@ public static AmazonS3Client CreateClient(IAWSS3BucketClientOptions options)
         else if (!string.IsNullOrWhiteSpace(options.AccessKey))
         {
             // AccessSecret can be empty.
-            Guard.NotNullOrWhiteSpace(options.Region, nameof(options.Region));
+            Guard.NotNullOrWhiteSpace(options.Region!, nameof(options.Region));
             RegionEndpoint region = RegionEndpoint.GetBySystemName(options.Region);
```

Comment:
The assertion should be on this line by the looks of things.

Reply:
we guard for the settings.region and not the created region. So this should be ok